pkg/pagerduty/pagerduty.go

```diff
@@ -2,112 +2,154 @@ package pagerduty
+		client: pd,
+	}, nil
+
 }
 
-// GetClusterIDFromAlertList retrieves the cluster ID from a list of PagerDuty alerts,
-// ensuring they have the same cluster ID or return an error if inconsistent or no alerts found.
-func (c *RealPagerDutyClient) GetClusterIDFromAlertList(alertList *pagerduty.ListAlertsResponse) (string, error) {
-	switch len(alertList.Alerts) {
-	case 0:
-		return "", fmt.Errorf("no alerts found for the given incident ID")
-	case 1:
-		clusterID, err := c.GetClusterIDFromAlert(&alertList.Alerts[0])
-		if err != nil {
-			return "", err
+// GetIncidentAlerts returns all the alerts belonging to a particular incident.
+func (pd *PagerDuty) GetIncidentAlerts(incidentID string) ([]Alert, error) {
+	var alerts []Alert
+
+	// Fetch alerts related to an incident via pagerduty API
+	incidentAlerts, err := pd.client.ListIncidentAlerts(incidentID)
+
+	if err != nil {
+		var aerr pdApi.APIError
+
+		if errors.As(err, &aerr) {
+			if aerr.RateLimited() {
+				return nil, fmt.Errorf("API rate limited")
+			}
+
+			return nil, fmt.Errorf("status code: %d, error: %s", aerr.StatusCode, err)
 		}
-		return clusterID, nil
-	default:
-		prevClusterID, err := c.GetClusterIDFromAlert(&alertList.Alerts[0])
+	}
+
+	for _, alert := range incidentAlerts.Alerts {
+		currentAlert := alert
+		formatAlert, err := pd.formatAlert(&currentAlert)
+
 		if err != nil {
-			return "", err
-		}
-		// Check if all alerts in the list have the same cluster ID
-		for _, alert := range alertList.Alerts {
-			currentAlert := alert
-			if currentClusterID, err := c.GetClusterIDFromAlert(&currentAlert); err != nil {
-				return "", err
-			} else if currentClusterID != prevClusterID {
-				return "", fmt.Errorf("not all alerts have the same cluster ID")
-			}
+			return nil, err
 		}
-		return prevClusterID, nil
+		alerts = append(alerts, formatAlert)
 	}
+	return alerts, nil
 }
 
-// GetClusterIDFromAlert extracts the cluster ID from a PagerDuty incident alert.
-// It expects the alert's body to have a Common Event Format.
-func (c *RealPagerDutyClient) GetClusterIDFromAlert(alert *pagerduty.IncidentAlert) (string, error) {
-	if alert == nil || alert.Body == nil {
-		return "", fmt.Errorf("given alert or it's body is empty")
-	}
+func (pd *PagerDuty) formatAlert(alert *pdApi.IncidentAlert) (formatAlert Alert, err error) {
+	formatAlert.IncidentID = alert.Incident.ID
 
-	cefDetails, ok := alert.Body["cef_details"].(map[string]interface{})
-	if !ok || cefDetails == nil {
-		return "", fmt.Errorf("missing or invalid Common Event Format Details of given alert")
-	}
+	formatAlert.Name = alert.Summary
+	formatAlert.Status = alert.Status
+	formatAlert.WebURL = alert.HTMLURL
 
-	detailsValue, ok := cefDetails["details"]
-	if !ok || detailsValue == nil {
-		return "", fmt.Errorf("missing or invalid 'details' field in Common Event Format Details")
-	}
+	// Check if the alert is of type 'Missing cluster'
+	isCHGM := alert.Body["details"].(map[string]interface{})["notes"]
```

Comment:
how can we identify it is CHGM with the `notes`?

Reply:
I added test case to cover that 
https://github.com/openshift/backplane-cli/pull/450/files#diff-957dde9e5fea16f1b4fda8327f293e8ffb26fb91d23fd369541d8af4ec320e0eR209-R219